src/distribution/chi_squared.rs

```diff
@@ -366,9 +415,112 @@ mod test {
+        test_almost(4.0, 2.270362845461478, 1e-13, |x| x.entropy());
+        test_almost(16.0, 3.108818195936093, 1e-13, |x| x.entropy());
+        test_almost(100.0, 4.061397128938097, 1e-13, |x| x.entropy());
+    }
+
+    #[test]
+    fn test_cdf() {
+        test_case(0.5, 0.0, |x| x.cdf(0.0));
+        test_almost(0.5, 5.165553208304657e-01, 1e-15, |x| x.cdf(0.1));
+        test_almost(0.5, 8.464864041916774e-01, 1e-15, |x| x.cdf(1.0));
+        test_almost(0.5, 9.995079748755537e-01, 1e-15, |x| x.cdf(10.0));
+        test_case(0.5, 1.0, |x| x.cdf(100.0));
+
+        test_case(1.0, 0.0, |x| x.cdf(0.0));
+        test_almost(1.0, 2.481703659541508e-01, 1e-15, |x| x.cdf(0.1));
+        test_almost(1.0, 6.826894921370859e-01, 1e-15, |x| x.cdf(1.0));
+        test_almost(1.0, 9.984345977419975e-01, 1e-15, |x| x.cdf(10.0));
+        test_case(1.0, 1.0, |x| x.cdf(100.0));
+
+        test_case(3.5, 0.0, |x| x.cdf(0.0));
+        test_almost(3.5, 3.184413836670893e-03, 1e-16, |x| x.cdf(0.1));
+        test_almost(3.5, 1.355315036184170e-01, 1e-15, |x| x.cdf(1.0));
+        test_almost(3.5, 9.719601382856896e-01, 1e-15, |x| x.cdf(10.0));
+        test_case(3.5, 1.0, |x| x.cdf(100.0));
+
+        test_case(100.0, 0.0, |x| x.cdf(0.0));
+        test_almost(100.0, 2.181059214078525e-32, 1e-42, |x| x.cdf(10.0));
+        test_almost(100.0, 6.953305247616201e-06, 1e-18, |x| x.cdf(50.0));
+        test_almost(100.0, 5.188083154720433e-01, 1e-13, |x| x.cdf(100.0));
+
+        test_almost(400.0, 2.024759014847327e-57, 1e-72, |x| x.cdf(100.0));
+        test_almost(400.0, 9.995177872404065e-01, 1e-15, |x| x.cdf(500.0));
+        test_case(400.0, 1.0, |x| x.cdf(1000.0));
+    }
+
+    #[test]
+    fn test_pdf() {
+        test_case(1.0, 0.0, |x| x.pdf(0.0));
+        test_almost(1.0, 1.2000389484301359798, 1e-15, |x| x.pdf(0.1));
+        test_almost(1.0, 0.24197072451914334980, 1e-15, |x| x.pdf(1.0));
+        test_almost(1.0, 0.010874740337283141714, 1e-15, |x| x.pdf(5.5));
+        test_almost(1.0, 4.7000792147504127122e-26, 1e-38, |x| x.pdf(110.1));
+        test_case(1.0, 0.0, |x| x.pdf(f64::INFINITY));
+        test_case(2.0, 0.0, |x| x.pdf(0.0));
+        test_almost(2.0, 0.47561471225035700455, 1e-15, |x| x.pdf(0.1));
+        test_almost(2.0, 0.30326532985631671180, 1e-15, |x| x.pdf(1.0));
+        test_almost(2.0, 0.031963930603353786351, 1e-15, |x| x.pdf(5.5));
+        test_almost(2.0, 6.1810004550085248492e-25, 1e-37, |x| x.pdf(110.1));
+        test_case(2.0, 0.0, |x| x.pdf(f64::INFINITY));
+        test_case(2.5, 0.0, |x| x.pdf(0.0));
+        test_almost(2.5, 0.24812852712543073541, 1e-15, |x| x.pdf(0.1));
+        test_almost(2.5, 0.28134822576318228131, 1e-15, |x| x.pdf(1.0));
+        test_almost(2.5, 0.045412171451573920401, 1e-15, |x| x.pdf(5.5));
+        test_almost(2.5, 1.8574923023527248767e-24, 1e-36, |x| x.pdf(110.1));
+        test_case(2.5, 0.0, |x| x.pdf(f64::INFINITY));
+        // test_case(f64::INFINITY, 0.0, |x| x.pdf(0.0));
```

Comment:
What does `pdf` return for `freedom == f64::INFINITY` normally? I'd like to have this case defined but am not necessarily sure we need an explicit branch in the code

Reply:
It returns a `NaN`. I guess as a result of `0 * Inf` when computing the pdf.

TBH I don't like the idea of returning `0` instead of `NaN` because it doesn't make sense for me... Shouldn't a pdf integrate to `1` over support? How can it be upheld when `pdf()` returns `0` everywhere?

Reply:
That's a good point. I think maybe being explicit about the `NaN` would be good, like adding a specific branch and note it in the docstring